src/wallet/sendTransaction.ts

```diff
@@ -0,0 +1,353 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { EventEmitter } from 'events';
+import walletApi from './api/walletApi';
+import MineTransaction from './mineTransaction';
+import HathorWalletServiceWallet from './wallet';
+import Transaction from '../models/transaction';
+import Output from '../models/output';
+import Input from '../models/input';
+import Address from '../models/address';
+import { HATHOR_TOKEN_CONFIG } from '../constants';
+import { shuffle } from 'lodash';
+import { SendTxError, UtxoError, WalletError, WalletRequestError } from '../errors';
+import { OutputRequestObj, InputRequestObj } from './types';
+
+type optionsType = {
+  outputs?: OutputRequestObj[],
+  inputs?: InputRequestObj[],
+  changeAddress?: string | null,
+  transaction?: Transaction | null,
+};
+
+class SendTransactionWalletService extends EventEmitter {
```

Comment:
Maybe we should rename the file to `sendTransactionWalletService.ts` or move it to `wallet-service/sendTransaction.ts`. What do you think?

Reply:
I agree with that if we decide to move the files in `src/new/` -> `src/wallet/`, otherwise I see no benefits on doing that. Should we create an issue to migrate the old facade to this new folder and migrate the code to typescript? I believe we will need to continue having both facades running, right?

Reply:
I guess we will need both facades from a long time. So I agree that we should refactor these files in the future. This change won't be compatible and will require update from use cases.

My points about renaming the file is that one might think that `SendTransactionWalletService` is the only one available since it is looking into the generic file `sendTransaction.ts`. Maybe we should move the code to `sendTransactionWalletService.ts` and keep just the export in the `sendTransaction.ts`. So, in the future, we can have both implementation available from the same file (e.g. `import { sendTransactionFullNode } from '@hathor/wallet-lib/wallet/sendTransaction';`).

Reply:
Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/221/commits/fa74421d218315e450eb1f4a542f127eae22ac9c